client/src/services/fridge/fridge.service.ts

```diff
@@ -1,43 +1,37 @@
-import { Injectable } from '@angular/core';
+import { Inject, Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { Observable } from 'rxjs';
 
 export interface Ingredient {
+  id: number;
   name: string;
 }
 
-//Will be replaced with API, using for testing
-export const INGREDIENTS: Ingredient[] = [
-  {name: 'Really long ingredient name' },
-  {name: 'Chicken' },
-  {name: 'Rice' },
-  {name: 'Apple' },
-  {name: 'Lettuce' },
-  {name: 'Beans' },
-  {name: 'Potato' },
-  {name: 'Salt' },
-  {name: 'Sugar' },
-  {name: 'Flour' },
-  {name: 'Eggs' },
-  {name: 'Pepper' },
-];
-
-
 @Injectable({
   providedIn: 'root',
 })
 export class FridgeService {
-  constructor() {}
 
-  public getAvailableIngredients(): string[] {
-    //TODO: Replace when Backend API available
-    return [...INGREDIENTS].map((ingredient) => ingredient.name).sort();
+  public apiIngredientsList: Ingredient[] = [];
+
+  constructor(private http: HttpClient, @Inject('BASE_URL') private baseUrl: string) {
```

Comment:
We have http-client.service set up that handles all the httpClient calls. You should be able to replace this with the service instead. Check recipe-service for reference.

Reply:
Updated to use http-client service!